### ISSUE TITLE
extract command result with many 0 bytes files (almost all source files)

### ISSUE THREAD

**Issue by esakal** [electron/asar#153]
Hi,
my environment: 
OS: MacOS high sierra 10.13.4, 
node: v8.9.4

I'm trying to extract asar file as followed:
```
$ npm install -g asar
$ cd application-name.app/Contents/Resources
$ mkdir code
$ asar extract app.asar code
```

it ends without errors and I have files in the `code` folder but most of the files are zero bytes like shown below:
![image](https://user-images.githubusercontent.com/4751797/43566320-aee3e062-9636-11e8-8d7e-8cf4be069e43.png)

Please assist,
Thanks
Eran.

**Comment by payitforwardnow** [comment]
Hi Eran, I've seen this issue before. Is the asar encrypted or protected in any way? I learned recently that there is something called "hidden errors". This happened to me when I tried to tweak the code to handle transforms on extract. Never saw an error but there was in fact an error during the extract process because I experienced the same 0 byte length files. Based on my experience, it appears the directory structure and 0 byte files (placeholders) are created first and the files don't receive any data unless the extract process completes successfully. Assuming the files weren't packed with 0 byte length (probably a reasonable assumption) there was probably some kind of error that occurred but is "hidden". I've tried the same process on a few electron apps with no issues. The Console app may provide system logs that could be helpful.
